### DISCUSSION TITLE
Add an ISO country code to the geofences table

### DISCUSSION THREAD

**Discussion by Ed-M72** [teslamate-org/teslamate#977] [Ideas]
**What would you like to be added**:
Please add an extra column in the geofences table to support the two character ISO country code.

**Why is this needed**:
I am building a custom dashboard for chargers, top chargers and cost to charger per month per charger.  I would like to break this down by country code or at least allow people to filter by country.

With the current implementation it shows me all chargers in GB and FR as that is what I have loaded.

**Additional context**:
I have added an additional column in the geofences table to support me, however it is not clear what happens on a version upgrade?  In case you would rather not deliver this could you please help me understand how customer modifications are not overridden on upgrade, e.g. naming standards for customers enhancement, e.g. z_country_code.

**Comment by adriankumpf** [comment]
> I have added an additional column in the geofences table to support me, however it is not clear what happens on a version upgrade? In case you would rather not deliver this could you please help me understand how customer modifications are not overridden on upgrade, e.g. naming standards for customers enhancement, e.g. z_country_code.

Some upgrades will change the database schema by adding, removing or renaming columns (see [migrations](https://github.com/adriankumpf/teslamate/tree/master/priv/repo/migrations)). If you change the table structure, it will basically be out of sync, which may or may not cause problems with future updates. if you just want to add this one column to _your_ install you will probably be fine but keep in mind that there may be problems with an update.

**Comment by Ed-M72** [comment]
Thats brilliant thank you Adrian.  I had a look at the migrations directory and now understand the structure.  I am not really "au fait" with GitHub or the approval process though which is definitely a downside.  I will try and upload a new migration file for comment and approval though and will reach out to you directly if I need more help.
